static/src/javascripts/lib/url.js

```diff
@@ -1,25 +1,29 @@
 // @flow
 
 import { hasPushStateSupport } from 'lib/detect';
+import memoize from 'lodash/functions/memoize';
 
 const supportsPushState = hasPushStateSupport();
 
 // returns "foo=bar&fizz=buzz" (eg. no ? symbol)
 const getCurrentQueryString = (): string =>
     window.location.search.replace(/^\?/, '');
 
+const queryStringToUrlVars = memoize(
+    (queryString: string): Object =>
+        Array.from(new URLSearchParams(queryString).entries()) // polyfill.io guarantees URLSearchParams
+            .reduce((acc, pair) => {
```

Comment:
We could [destructure](https://stackoverflow.com/questions/36961080/syntax-for-destructuring-arrays-into-function-parameters-in-es6#36961081) the `pair` array to clarify the logic here:

```
(acc, [key, value]) => {
  acc[key] = pair[value] === '' ? true : value;
  return acc;
}
```

Reply:
Beautiful